File: Data Structures & Algorithms/house-robber/submission-1.java
class Solution {
    int rb( int i, int[] nums,  int[] dp){
        if (i>= nums.length){
            return 0;
        }
        if (dp[i] != -1){
            return dp[i];
        }
        return dp[i] = Math.max( nums[i] + rb(i+2,nums,dp) , rb(i+1,nums,dp));
    }
    public int rob(int[] nums) {
        int n=nums.length;
        int []dp = new int[n];
        Arrays.fill(dp,-1);
        return rb(0,nums,dp);
        
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        p = [0] * len(nums)
        p[0] = nums[0]
        for i in range(1,len(nums)):
            curr = p[i-1] * nums[i]
            p[i] = max(curr,nums[i])
        return max(p)
        